messaginginapp/src/main/java/io/customer/messaginginapp/type/InAppMessage.kt

```diff
@@ -0,0 +1,23 @@
+package io.customer.messaginginapp.type
+
+import build.gist.data.model.GistMessageProperties
+import build.gist.data.model.Message
+
+data class InAppMessage(
+    val instanceId: String,
+    val messageId: String,
+    val deliveryId: String? // (Currently taken from Gist's campaignId property). Can be null when sending test in-app messages
+) {
+    companion object {
+        internal fun getFromGistMessage(gistMessage: Message): InAppMessage {
```

Comment:
You already created `Message` extension earlier, do you think it would make more sense if we add it there? 

```kotlin
fun Message.getFromGistMessage(): InAppMessage {
    val gistProperties = GistMessageProperties.getGistProperties(this)
    val campaignId = gistProperties.campaignId

    return InAppMessage(
        instanceId = this.instanceId,
        messageId = this.messageId,
        deliveryId = campaignId
    )
}
```

Reply:
I do enjoy extensions and see value in creating a `fun Message.getFromGistMessage(): InAppMessage` extension function. 

However, I do like encapsulating the construction of a `InAppMessage` object existing inside of the `InAppMessage` object. 

If we want to leverage extensions, I vote we keep `InAppMessage.getFromGistMessage()` as-is but create an extension: 

```kt
fun Message.getFromGistMessage() = InAppMessage.getFromGistMessage(this)
```